src/jvm/test/org/jetbrains/kotlinx/lincheck_test/trace_debugger/RandomTests.kt

```diff
@@ -0,0 +1,388 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.trace_debugger
+
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.isInTraceDebuggerMode
+import org.jetbrains.kotlinx.lincheck_test.representation.BaseTraceRepresentationTest
+import org.junit.Assume.assumeTrue
+import org.junit.Before
+import java.util.concurrent.ThreadLocalRandom
+import java.util.concurrent.atomic.DoubleAccumulator
+import java.util.concurrent.atomic.DoubleAdder
+import java.util.concurrent.atomic.LongAccumulator
+import java.util.concurrent.atomic.LongAdder
+import kotlin.random.Random
+import kotlin.random.nextInt
+import kotlin.random.nextLong
+import kotlin.random.nextUBytes
+import kotlin.random.nextUInt
+import kotlin.random.nextULong
+import java.util.Random as JRandom
+
+abstract class RandomTests : AbstractDeterministicTest() {
+    override val alsoRunInLincheckMode: Boolean get() = true
+}
+
+class RandomInt1Test : RandomTests() {
+    @Operation
+    fun operation() = Random.Default.nextInt()
+}
+
+class RandomInt2Test : RandomTests() {
+    @Operation
+    fun operation() = Random.Default.nextInt(1000)
+}
+
+class RandomInt3Test : RandomTests() {
+    @Operation
+    fun operation() = Random.Default.nextInt(1000, 10000)
+}
+
+class RandomInt4Test : RandomTests() {
+    @Operation
+    fun operation() = Random.Default.nextInt(1000..10000)
+}
+
+class RandomInt5Test : RandomTests() {
+    @Operation
+    fun operation() = runCatching { Random.Default.nextInt(10000, 1000) }.exceptionOrNull()!!.message
+}
+
+class NotRandom : Random() {
```

Comment:
Why not just use regular `Random` with fixed seed?

Reply:
To check the random more radically: with a custom random.